### PR TITLE
Feat/json view

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

So the user has the same UX in payload rendering (Svelte/Flutter/Dashboard) the temporary solution (see mockFlutterJSONFormatting) was replaced with Json component.

# Note

All nodes are initially expanded to have the same state as in dashboard (currently in discussion with Mischa)

# Changes

- Used Json component in ProposalActions

# Tests

- Json test

# Screenshots

## Svelte <--> Flutter
Screenshot 2022-05-06 at 11 12 24

https://user-images.githubusercontent.com/98811342/167110301-85bec5ee-6181-47ea-b6ed-8796b2575e3a.mov

### CODE REVIEW

File: frontend/svelte/src/tests/lib/components/proposal-detail/ProposalDetailCard/ProposalActions.spec.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ describe("ProposalActions", () => {
     for (const [key, value] of fields) {
       const element = getByText(key);
       expect(element).toBeInTheDocument();
-      expect(element.nextElementSibling?.textContent).toBe(value);
+      if (typeof value !== "object") {
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/svelte/src/tests/lib/components/proposal-detail/ProposalDetailCard/ProposalActions.spec.ts:41]: Good point. To be more reliable let's check for rendered json.

File: frontend/svelte/src/lib/components/common/Json.svelte [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    | "function"
+    | "null"
+    | "number"
+    | "object"
+    | "principal"
+    | "string"
+    | "symbol"
+    | "undefined";
+  const valueType = (value): ValueType => {
+    if (value === null) return "null";
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/common/Json.svelte:23]: can it be `undefined`?

File: frontend/svelte/src/lib/components/common/Json.svelte [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+  let collapsed: boolean = true;
+  $: collapsed =
+    _collapsed === undefined ? defaultExpandedLevel < _level : _collapsed;
+
+  const toggle = () => (collapsed = !collapsed);
+</script>
+
+{#if isExpandable && hasChildren}
+  {#if collapsed}
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/common/Json.svelte:82]: that's quite multiple lines of code duplicated, can we spare the duplication?

File: frontend/svelte/src/lib/components/proposal-detail/ProposalDetailCard/ProposalActions.svelte [hunk 4]
```diff
@@ -5,24 +5,33 @@
...
-<CardBlock>
+<CardBlock limitHeight={false}>
   <svelte:fragment slot="title">{actionKey}</svelte:fragment>
   <ul>
     {#each actionFields as [key, value]}
       <li>
         <h4>{key}</h4>
-        <p>{value}</p>
+        {#if typeof value === "object"}
+          <p class="json">
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/proposal-detail/ProposalDetailCard/ProposalActions.svelte:27]: the `Json` contains `ul` which semantically - if I am not wrong - should not be children of `p`

**Review by peterpeterparker — COMMENTED** [review]
Looks cool! Added small comments @mstrasinskis

**Pull request merged**